components/eam/src/physics/cam/ndrop.F90

```diff
@@ -1084,20 +868,349 @@ subroutine dropmixnuc( &
+   real(r8) :: frac_delt_cld        ! fractional change in cloud fraction [fraction]
+   integer  :: imode       ! mode counter variable
+   integer  :: lspec       ! species counter variable
+   integer  :: mm                  ! local array index for MAM number, species
+   real(r8) :: dact             ! cloud-borne aerosol tendency due to cloud frac tendency [#/kg or kg/kg]
+   real(r8) :: fn(ntot_amode)              ! activation fraction for aerosol number [fraction]
+   real(r8) :: fm(ntot_amode)              ! activation fraction for aerosol mass [fraction]
+   integer  :: num_idx  ! number index
+   integer  :: spc_idx  ! species index
+   real(r8) :: flux_fullact ! flux of activated aerosol fraction assuming 100% activation [m/s]
+   real(r8) :: fluxn(ntot_amode)     ! flux of activated aerosol number fraction into cloud [m/s]
+   real(r8) :: fluxm(ntot_amode)     ! flux of activated aerosol mass fraction into cloud [m/s]
+
+
+         ! shrinking cloud ......................................................
+         !    treat the reduction of cloud fraction from when cldn(i,k) < cldo(i,k)
+         !    and also dissipate the portion of the cloud that will be regenerated
+
+         if (cldn_in < cldo_in) then
+            !  droplet loss in decaying cloud
+            !++ sungsup
+            nsource_out = nsource_out + qcld_in*(cldn_in - cldo_in)/cldo_in*dtinv
+            qcld_in      = qcld_in*(1._r8 + (cldn_in - cldo_in)/cldo_in)
+            !-- sungsup
+
+            ! convert activated aerosol to interstitial in decaying cloud
+
+            delt_cld = (cldn_in - cldo_in)
+            frac_delt_cld = delt_cld / cldo_in
+            do imode = 1, ntot_amode
+               mm = mam_idx(imode,0)
+               dact   = raercol_cw_pt(mm)*frac_delt_cld
+               raercol_cw_pt(mm) = raercol_cw_pt(mm) + dact   ! cloud-borne aerosol
+               raercol_pt(mm)    = raercol_pt(mm) - dact
+               do lspec = 1, nspec_amode(imode)
+                  mm = mam_idx(imode,lspec)
+                  dact    = raercol_cw_pt(mm)*frac_delt_cld
+                  raercol_cw_pt(mm) = raercol_cw_pt(mm) + dact  ! cloud-borne aerosol
+                  raercol_pt(mm)    = raercol_pt(mm) - dact
+               enddo
+            enddo
+         endif   ! cldn(icol,kk) < cldo(icol,kk)
+
+         ! growing cloud ......................................................
+         !    treat the increase of cloud fraction from when cldn(i,k) > cldo(i,k)
+         !    and also regenerate part of the cloud
+
+         if (cldn_in-cldo_in > 0.01_r8) then
+
+            call get_activate_frac(  &
+                 state_q_in, &
+                 cs_kk_in, cs_kk_in,    &
+                 wtke_in, temp_in,    &
+                 fn, fm, fluxn,         &
+                 fluxm,flux_fullact)
+
+!  store for output activation fraction of aerosol
+            factnum_out(:) = fn
+            delt_cld = (cldn_in - cldo_in)
```

Comment:
Both delt_cld and frac_delt_cld have been moved to the top of the subroutine, outside of the if-then structures.

Reply:
Actually, it was a mistake to move frac_delt_cld outside the if-then structures, because it is undefined when cldo_in = 0, but if cldn_in < cldo_in, this computation is safe.

Reply:
Good point, I guess moving just the `delt_cld` variable should be okay.